src/lwc/geFormField/geFormField.js

```diff
@@ -384,19 +405,70 @@ export default class GeFormField extends LightningElement {
 
         }
 
-        lookup.setSelected({value, displayValue});
-
+        if (this.sourceFieldAPIName === DONATION_RECORD_TYPE_NAME.fieldApiName) {
+            lookup.setSelected({value: value});
+        } else {
+            lookup.setSelected({value, displayValue});
+        }
     }
 
-
     @api
     reset() {
         this.value = this._defaultValue;
 
         if (this.isLookup) {
             const lookup = this.template.querySelector('c-ge-form-field-lookup');
             lookup.reset();
+            if (this.fieldApiName === 'RecordTypeId') {
+                // Using setTimeout here ensures that this recordTypeId
+                // will be set on sibling fields after they are reset by queueing the event.
+                setTimeout(() => {
```

Comment:
This is interesting ... I kind of reflexively want to avoid setTimeout() but this looks like an interesting technique to address the potential order of execution issue

Reply:
Yeah, it feels a bit strange but seems to be an acceptable practice to resolve race conditions.  In this case, it is the only way I could manage for picklists to get set back to the correct recordTypeId after the reset() flow independent of the order of the fields on the form.